### ISSUE TITLE
Some languages available on google translate that are not working

### ISSUE THREAD

**Issue by stephen147** [UweKeim/ZetaResourceEditor#46]
I've successfully translated 61 languages (I'll put the list below for anyone who wants a template file setup) but here's a list of the languages that are not working in Zeta but are on Google Translate.

There's also some that don't even get shown in Zeta but are on Google Translate shown below.

# Not getting shown in Zeta but on Google
```
Lang.fil-Latn (Filipino).resx
Lang.ky-Cyrl (Kyrgyz).resx
Lang.mi-Latn (Maori).resx
```

There are a few important ones in there like German, Finnish, Italian, Japanese and Polish.

# Files that are not getting translated in Zeta
```
Lang.ca (Catalan Spanish).resx
Lang.cs (Czech).resx
Lang.da (Danish).resx
Lang.de (German).resx
Lang.de-de (German).resx
Lang.et (Estonian).resx
Lang.eu (Basque).resx
Lang.fa (Persian).resx
Lang.fi (Finnish).resx
Lang.ga (Irish).resx
Lang.gd-Latn (Scottish Gaelic).resx
Lang.gl (Galician).resx
Lang.gu (Gujarati).resx
Lang.ha-Latn (Hausa (Latin)).resx
Lang.he (Hebrew).resx
Lang.hi (Hindi).resx
Lang.hr (Croatian).resx
Lang.hu (Hungarian).resx
Lang.hy (Armenian).resx
Lang.id (Indonesian).resx
Lang.ig-Latn (Igbo).resx
Lang.is (Icelandic).resx
Lang.it (Italian).resx
Lang.it-it (Italian).resx
Lang.ja (Japanese).resx
Lang.ka (Georgian).resx
Lang.kk (Kazakh).resx
Lang.km (Khmer).resx
Lang.kn (Kannada).resx
Lang.ko (Korean).resx
Lang.lb (Luxembourgish).resx
Lang.lt (Lithuanian).resx
Lang.lv (Latvian).resx
Lang.mr (Marathi).resx
Lang.ms (Malay).resx
Lang.pl (Polish).resx

```

# Working via Googles API so you can use this as a file template
```
Lang.af.resx
Lang.am.resx
Lang.ar-sa.resx
Lang.az-Latn.resx
Lang.be.resx
Lang.bg.resx
Lang.bn-BD.resx
Lang.bn-IN.resx
Lang.bs.resx
Lang.cy.resx
Lang.el.resx
Lang.en-GB.resx
Lang.en-US.resx
Lang.es-ES.resx
Lang.es-MX.resx
Lang.es-US.resx
Lang.es.resx
Lang.fr-CA.resx
Lang.fr-FR.resx
Lang.fr.resx
Lang.mk.resx
Lang.ml.resx
Lang.mn-Cyrl.resx
Lang.mt.resx
Lang.ne.resx
Lang.nl-BE.resx
Lang.nl.resx
Lang.or.resx
Lang.prs-Arab.resx
Lang.pt-BR.resx
Lang.pt-PT.resx
Lang.qut-Latn.resx
Lang.ro.resx
Lang.ru.resx
Lang.rw.resx
Lang.sd-Arab.resx
Lang.si.resx
Lang.sk.resx
Lang.sl.resx
Lang.sq.resx
Lang.sr-Cyrl-BA.resx
Lang.sr-Cyrl-RS.resx
Lang.sr-Latn-RS.resx
Lang.sv.resx
Lang.sw.resx
Lang.ta.resx
Lang.te.resx
Lang.tg-Cyrl.resx
Lang.th.resx
Lang.tk-Latn.resx
Lang.tr.resx
Lang.tt-Cyrl.resx
Lang.ug-Arab.resx
Lang.uk.resx
Lang.ur.resx
Lang.uz-Latn.resx
Lang.vi.resx
Lang.xh.resx
Lang.yo-Latn.resx
Lang.zh-Hans.resx
Lang.zh-Hant.resx
Lang.zu.resx
```

**Comment by UweKeim** [comment]
Do you consider this to be a bug in the application? Or is your issue here more of an informative kind?

**Comment by stephen147** [comment]
Yes, I do. It's the two sections below. They're the languages that are on Google Translate but don't work in Zeta.

1. Files that are not getting translated in Zeta
2. Not getting shown in Zeta but on Google

The others are just informative.

**Comment by UweKeim** [comment]
Probably the reason for both parts is this:

After getting the list of possible languages from a translation provider (Bing/Azure or Google), I check each of the languages against the currently available languages on the local Windows system.

See e.g. [the implementation of the `IsSupportedLanguage` method](https://github.com/UweKeim/ZetaResourceEditor/blob/master/Source/RuntimeBusinessLogic/Translation/TranslationHelper.cs#L41).

Online those languages that can be mapped onto an existing Windows/.NET language are being made available.

I've decided this long time ago and am not sure how removing this behavior would affect the different parts of the application.

You are of course free to submit a pull request, if you could "fix" this.

**Comment by stephen147** [comment]
I was looking here at the: [Supported Culture Codes](https://docs.microsoft.com/en-us/bingmaps/rest-services/common-parameters-and-types/supported-culture-codes).

I'm only starting out programming in C# so I doubt I'd have the ability to fix this.

If the program is searching the local windows system with the IsSupportedLanguage method, does that make sense seeing as in development not all systems are the same?

**Comment by UweKeim** [comment]
Yes, in my opinion that does make sense, since I could not get a `CultureInfo` from this language object otherwise.

**Comment by stephen147** [comment]
Oh right, I thought it solely relied on using the google or bing API to get the languages that are supported in their APIs'.

**Comment by stephen147** [comment]
There must be something happening as I'm getting some of the languages that failed previously. I'm using the same dev machine here.

See attachment of the Galician language getting translated.

![image](https://user-images.githubusercontent.com/14054664/142769674-f3cc1feb-8570-4881-aba2-b42d9d8c480b.png)

But after the Galician language here it shows this message.

![image](https://user-images.githubusercontent.com/14054664/142769711-8d517829-2cac-47bf-b79b-250f3a14c1bb.png)

Then if I click okay and choose the next language Gujarati it works for that and the same message will appear after that.

The same message will appear about the deleted row info. If I repeat this for each of remaining languages in turn then it works.

Once I'm complete I'll zip the `.resx` files here for you to test.
